PostmarkInboundWebhookMailParser/PostmarkIncomingMessage.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Newtonsoft.Json;
+
+namespace PostmarkInboundWebhookMailParser
+{
+    public class PostmarkIncomingMessage
+    {
+        public string FromName { get; set; }
+        public string MessageStream { get; set; }
+        //public string From { get; set; }
+        public EmailFull FromFull { get; set; }
+        //public string To { get; set; }
+        public IList<EmailFull> ToFull { get; set; }
+        //public string Cc { get; set; }
+        public IList<EmailFull> CcFull { get; set; }
+        //public string Bcc { get; set; }
+        public IList<EmailFull> BccFull { get; set; }
+        public string OriginalRecipient { get; set; }
+        public string Subject { get; set; }
+        [JsonProperty("MessageID")]
+        public string MessageId { get; set; }
+        public string ReplyTo { get; set; }
+        public string MailboxHash { get; set; }
+        public string Date { get; set; }
+        public string TextBody { get; set; }
+        public string HtmlBody { get; set; }
+        public string StrippedTextReply { get; set; }
+        public string Tag { get; set; }
+        public IList<Header> Headers { get; set; }
+        public IList<Attachment> Attachments { get; set; }
+
+        private bool? _isBodyHtml;
+        public bool IsBodyHtml
+        {
+            get
+            {
+                if (_isBodyHtml == null)
+                {
+                    _isBodyHtml = !string.IsNullOrWhiteSpace(HtmlBody);
+                }
+
+                return _isBodyHtml ?? false;
+            }
+        }
+
+        private DateTime? _parsedDate;
+        /// <summary>
+        /// Returns the JSON string date properly parsed into <see cref="DateTime"/>. Could potentially adapt via <see cref="JsonSerializerSettings"/> <see cref="DateParseHandling"/>, but that brings in a whole new set of complications.
+        /// </summary>
+        public DateTime ParsedDate
+        {
+            get
+            {
+                if (_parsedDate == null)
+                {
+                    _parsedDate = DateTime.Parse(this.Date, CultureInfo.InvariantCulture, DateTimeStyles.AdjustToUniversal | DateTimeStyles.AssumeUniversal);
+                }
+
+                // Can safely assume that date is always set so never null.
+                return _parsedDate.Value;
+            }
+        }
+
+        //public void AttachAttachments()
```

Comment:
I never like to see code commented out. That's whey we have source control

Reply:
This is not commented out code so much as reference for how to use in context. Yes, could be put in a more applicable part (for instance it is now referenced in the README). At the time I didn't want to be doing a wholesale renovation of everything.